### PR TITLE
Changelog for 5.2.1-preview

### PR DESCRIPTION
**Pull request by yitam**
This change is [Reviewable](https://reviewable.io/reviews/microsoft/msphpsql/779)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
...
+Updated PECL release packages. Here is the list of updates:
+
+### Added
+- Added support for Azure Key Vault for Always Encrypted for basic CRUD functionalities 
+- Added support for macOS High Sierra (requires [MS ODBC Driver 17+](https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-2017))
+
+### Fixed
+- Issue [#577](https://github.com/Microsoft/msphpsql/issues/577) - Idle Connection Resiliency doesn't work with Column Encryption enabled connection
+- Issue [#678](https://github.com/Microsoft/msphpsql/issues/678) - Idle Connection Resiliency doesn't work with Connection Pooling bug
+- Issue [#699](https://github.com/Microsoft/msphpsql/issues/699) - binding output parameter failed when the query in the stored procedure returned no data. The test case has been added to the test lab.
```
Comment (david-puglielli) [inline CHANGELOG.md]: Capitalise the B

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
...
+- Issue [#705](https://github.com/Microsoft/msphpsql/issues/705) - AE - Retrieving a negative decimal value (edge case) as output parameter causes truncation
+- Issue [#706](https://github.com/Microsoft/msphpsql/issues/706) - AE - Cannot insert double with precision and scale (38, 38)
+- Issue [#707](https://github.com/Microsoft/msphpsql/issues/707) - AE - fetching decimals / numerics as output parameters bound to PDO::PARAM_BOOL or PDO::PARAM_INT returns floats, not integers
```
Comment (david-puglielli) [inline CHANGELOG.md]: Capitalise the F in fetching

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
...
+
+### Fixed
+- Issue [#577](https://github.com/Microsoft/msphpsql/issues/577) - Idle Connection Resiliency doesn't work with Column Encryption enabled connection
+- Issue [#678](https://github.com/Microsoft/msphpsql/issues/678) - Idle Connection Resiliency doesn't work with Connection Pooling bug
+- Issue [#699](https://github.com/Microsoft/msphpsql/issues/699) - binding output parameter failed when the query in the stored procedure returned no data. The test case has been added to the test lab.
+- Issue [#705](https://github.com/Microsoft/msphpsql/issues/705) - AE - Retrieving a negative decimal value (edge case) as output parameter causes truncation
+- Issue [#706](https://github.com/Microsoft/msphpsql/issues/706) - AE - Cannot insert double with precision and scale (38, 38)
+- Issue [#707](https://github.com/Microsoft/msphpsql/issues/707) - AE - fetching decimals / numerics as output parameters bound to PDO::PARAM_BOOL or PDO::PARAM_INT returns floats, not integers 
+- Issue [#735](https://github.com/Microsoft/msphpsql/issues/735) - PDO lastInsertId primary bigInt, after id 999999999 return ''
```
Comment (david-puglielli) [inline CHANGELOG.md]: This description is confusing - what has been changed?

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
...
+- Pull Request [#759](https://github.com/Microsoft/msphpsql/pull/759) - removed the limitation of binding a binary as inout param as PDO::PARAM_STR with SQLSRV_ENCODING_BINARY
+- Pull Request [#775](https://github.com/Microsoft/msphpsql/pull/775) - fixed the problem for output params with SQL types specified as SQLSRV_SQLTYPE_DECIMAL or SQLSRV_SQLTYPE_NUMERIC
```
Comment (david-puglielli) [inline CHANGELOG.md]: Capitalise the r and f

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
...
+
+### Limitations
+- No support for inout / output params when using sql_variant type
+- In Linux and macOS, setlocale() only takes effect if it is invoked before the first connection. Attempting to set the locale after connection will not work
+- Always Encrypted feature, which requires [MS ODBC Driver 17+](https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-2017)
+  - only Windows Certificate Store and Azure Key Vault are supported
+  - Issue [#716](https://github.com/Microsoft/msphpsql/issues/716) - Named Parameters in Sub Queries
```
Comment (david-puglielli) [inline CHANGELOG.md]: Give a brief description of the limitation here?

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
...
+  - [Always Encrypted limitations](https://docs.microsoft.com/en-us/sql/connect/php/using-always-encrypted-php-drivers?view=sql-server-2017#limitations-of-the-php-drivers-when-using-always-encrypted)
+
+### Known Issues
+- Connection pooling on Linux or macOS not recommended with [unixODBC](http://www.unixodbc.org/) < 2.3.6
+- When pooling is enabled in Linux or macOS
+  - unixODBC <= 2.3.4 (Linux and macOS) might not return proper diagnostics information, such as error messages, warnings and informative messages
+  - due to this unixODBC bug, fetch large data (such as xml, binary) as streams as a workaround. See the examples [here](https://github.com/Microsoft/msphpsql/wiki/Features#pooling)
+- With ColumnEncryption enabled, calling stored procedure with XML parameter does not work (Issue [#674](https://github.com/Microsoft/msphpsql/issues/674))
```
Comment (david-puglielli) [inline CHANGELOG.md]: ....calling stored procedures with XML parameters does not.....

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,7 +3,41 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/) 
 
-## Windows/Linux/macOS 5.2.0 - 2017-02-14
+## 5.2.1-preview - 2018-06-01
+Updated PECL release packages. Here is the list of updates:
+
+### Added
+- Added support for Azure Key Vault for Always Encrypted for basic CRUD functionalities
```
Comment (david-puglielli) [inline CHANGELOG.md]: I think we should mention that AE now supports Linux and mac through AKV

File: README.md [hunk 8]
```diff
@@ -96,15 +93,9 @@ The version number may have trailing pre-release version identifiers to indicate
 ## Guidelines for Reporting Issues
 We appreciate you taking the time to test the driver, provide feedback and report any issues.  It would be extremely helpful if you:
 
+- First check [FAQ](https://github.com/Microsoft/msphpsql/wiki/FAQ)
```
Comment (david-puglielli) [inline README.md]: the FAQ

File: README.md [hunk 9]
```diff
@@ -96,15 +93,9 @@ The version number may have trailing pre-release version identifiers to indicate
...
 - Report each issue as a new issue (but check first if it's already been reported)
-- Try to be detailed in your report. Useful information for good bug reports includes:
-  * What you are seeing and what the expected behaviour is
-  * Can you connect to SQL Server via `sqlcmd`? 
-  * Which driver: SQLSRV or PDO_SQLSRV?
-  * Environment details: e.g. PHP version, thread safe (TS) or non-thread safe (NTS), 32-bit or 64-bit?
-  * Table schema (for some issues, the data types make a big difference!)
-  * Any other relevant information you want to share
-- Try to include a PHP script demonstrating the isolated problem.
+- Please address the questions in the new issue template and provide scripts, table schema, and/or any detail that help reproducing the problem(s)
```
Comment (david-puglielli) [inline README.md]: ...any details that may help reproduce the problem(s)
